Validators

kennyqiyenkan opened the pull request:
Added the following:
- Validator Interface
- Validateable Interface
- ValidateableValidator Singleton
- Validated DAO

kennyqiyenkan committed:
Merge remote-tracking branch 'foam-framework/master'

kennyqiyenkan committed:
Merge remote-tracking branch 'foam-framework/master'

kennyqiyenkan committed:
Merge remote-tracking branch 'foam-framework/master'

kennyqiyenkan committed:
Validation files

kennyqiyenkan committed:
Merge remote-tracking branch 'foam-framework/master' into validators

kgrgreer commented on java_src/foam/core/Validateable.java:
```diff
@@ -0,0 +1,13 @@
+
+package foam.core;
+
+import java.lang.IllegalStateException;
```
You don't need to import anything from the java.lang package since it's already imported by default.

kgrgreer commented on java_src/foam/core/ValidateableValidator.java:
```diff
@@ -0,0 +1,22 @@
+package foam.core;
+
+import foam.core.*;
+import java.lang.IllegalStateException;
```
Not needed.

kgrgreer commented on java_src/foam/core/ValidateableValidator.java:
```diff
@@ -0,0 +1,22 @@
+import java.lang.IllegalStateException;
+
+public class ValidateableValidator {
+  private static ValidateableValidator sharedInstance = new ValidateableValidator();
```
renamed instance_

kgrgreer commented on java_src/foam/core/ValidateableValidator.java:
```diff
@@ -0,0 +1,22 @@
+public class ValidateableValidator {
+  private static ValidateableValidator sharedInstance = new ValidateableValidator();
+
+  public static ValidateableValidator getInstance() { return sharedInstance; }
```
rename instance()

kgrgreer commented on java_src/foam/core/ValidateableValidator.java:
```diff
@@ -0,0 +1,22 @@
+  private ValidateableValidator() {}
+
+  public static void validate(Object obj) throws IllegalStateException {
+    (Validateable)obj.validate();
```
((Validateable) obj).validate();

kgrgreer commented on java_src/foam/dao/ValidatedDAO.java:
```diff
@@ -0,0 +1,27 @@
+package foam.dao;
+
+import foam.core.Validator;
+
```
Remove blank line. You're importing both foam.core.* and foam.core.Validator.

kgrgreer commented on java_src/foam/dao/ValidatedDAO.java:
```diff
@@ -0,0 +1,27 @@
+  implements Validator
+{
+  public void validate(FObject obj) throws IllegalStateException {
+    (Validateable)obj.validate();
```
((Validateable) obj).validate();

kennyqiyenkan committed:
Merge remote-tracking branch 'foam-framework/master' into validators

kennyqiyenkan committed:
Made changes according to comments

removed IllegalStateException imports
wrapped Typecasts
renamed singleton variables

kennyqiyenkan committed:
Merge remote-tracking branch 'foam-framework/master' into validators
